[peterXGD] iP

peterXGD opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

MayDalag committed:
l0

MayDalag committed:
level-1

MayDalag committed:
level-2

MayDalag committed:
level 3

MayDalag committed:
level-4

MayDalag committed:
Level-5

MayDalag committed:
The real Level 5

MayDalag committed:
The real level 5 v2

MayDalag committed:
level 6 delete with collections implemented earlier

MayDalag committed:
enum-version

MayDalag committed:
Level-8 complete

MayDalag committed:
Level-6 v1

MayDalag committed:
Merge branch 'master' into branch-Level-7

* master:
  Level-6 v1

MayDalag committed:
Further fix needed until level-8
Fix bugs in level-7 to allow mark and unmark of files
Need to rewrite task files to accommodate time in them and modify main as well.

MayDalag committed:
Level-8 completed.

MayDalag committed:
Level-8 test

MayDalag committed:
Revert "Level-8 test"

This reverts commit 8b7b6285fd5f710579331ebaa008481f089e517a.

MayDalag committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Revert "Level-8 test"
  Level-8 test
  Level-8 completed.
  Further fix needed until level-8 Fix bugs in level-7 to allow mark and unmark of files Need to rewrite task files to accommodate time in them and modify main as well.
  Level-8 complete

MayDalag committed:
Finish Level-9

OOP is done by seperating out Parser, Storage, TaskList and Ui class in addition to existing ones.

MayDalag committed:
Master branch dummy

To be merged with branch-A-More-OOP

MayDalag committed:
Merge branch 'A-more-oop'

* A-more-oop:
  Finish Level-9

MayDalag committed:
Duke package: duke files added to duke package

MayDalag committed:
Duke package: refractor duke to venus for chatbot name

This is done as it makes more sense to package into venus package instead of duke

MayDalag committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  Bump gradle and lib version
  Add Gradle support
  docs/README.md: Tweak document template

MayDalag committed:
Add Gradle to current level increment

This is to comply with requirement.

We use Gradle which can help with jUnit tests and JAR creation later.

MayDalag committed:
Add JUnit test for some Parser and TimeFormatter and fix some bugs

This is done as part of requirement for JUnit test, and it fixes some bugs
during the testing process.

MayDalag committed:
Make Gradle produce JAR files with correct name

MayDalag committed:
Make file control OS independent

A mistake was made that makes java run on windows and macOS only.

It is corrected by using the FIle class in Java.

MayDalag committed:
Fix some formatting issues

This will make style check easier for future
tags.

MayDalag committed:
Correct codes to comply with coding standard

MayDalag committed:
Adds JavaDoc to non-trivial methods

MayDalag committed:
Change enum name to camel case

This is to follow naming convention.

MayDalag committed:
Fix error where todo is not named properly

MayDalag committed:
Add find method in TaskList

This is to add a find method required for the increment.

MayDalag committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add find method in TaskList

MayDalag committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Adds JavaDoc to non-trivial methods

MayDalag committed:
Merge branch 'branch-A-CodingStandard' to master branch

* branch-A-CodingStandard:
  Change enum name to camel case
  Correct codes to comply with coding standard

javierng2knus commented on src/main/java/venus/Deadline.java:
```diff
@@ -0,0 +1,54 @@
+     * @param s The deadline event itself
+     * @param deadline The time string used for the deadline in "YYYY-MM-DD" format.
+     */
+    public Deadline(String s, boolean mark, String deadline) throws DateTimeParseException {
```
It seems for this function, the name of the parameter `mark` is violating the naming convention [here](https://se-education.org/guides/conventions/java/intermediate.html#comments)

Perhaps consider changing it to `isMarked` or 'hasBeenMarked'?

Also I found other instances of this in the Task related classes (`Event`, `Todo`, `Task`)  👍

javierng2knus commented on src/main/java/venus/Parser.java:
```diff
@@ -0,0 +1,114 @@
+     * @param input Input to be parsed.
+     * @return String array of Deadline content and time (unformatted).
+     */
+    public static String[] findDeadlineContent(String input) throws DukeException {
```
It seems for this function, the `@throws` is missing, violating the commenting convention [here](https://se-education.org/guides/conventions/java/intermediate.html#comments)

Also I found other instances (listed below for your convenience) but most of the methods that can throw exceptions seems to include the `@throws` so good job!👍 

`findEventParts(String input) throws DukeException`
`Deadline(String s, boolean mark, String deadline) throws DateTimeParseException`

javierng2knus commented on src/main/java/venus/Task.java:
```diff
@@ -0,0 +1,51 @@
+     *
+     * @param item String representation of task.
+     */
+    public Task(String item){
```
It seems that you may have missed a white space after some of the statements across multiple files such as `public Task(String item){` which violates the layout convention [here](https://se-education.org/guides/conventions/java/intermediate.html#layout)

You can ctrl-f or whatever "find" command your system uses and search `){` to quickly find the locations needing white spaces 👍

javierng2knus reviewed (commented):
I like how you comprehensively included Javadoc and comments across your file! 😊
Other than those nitpicks, LGTM!👍

Edit: 
I realized that we allocated 2 specific people to review instead of just random choosing people...🤦 
Sincerest apologies for any inconvenience caused. 🙇

ashleyy2444 commented on src/main/java/venus/TimeFormatter.java:
```diff
@@ -0,0 +1,41 @@
+     * @return LocalDate of input.
+     * @throws DateTimeParseException Throws exception when invalid date format is used.
+     */
+    public static LocalDate stringToTime(String s) throws DateTimeParseException {
```
It seems that you violated the Java coding standards for the naming of this method. Names representing methods must be verbs and written in camelCase. You can replace this method name with 'convertStringToTime' instead.

ashleyy2444 commented on src/main/java/venus/Storage.java:
```diff
@@ -0,0 +1,136 @@
+    public void saveToFile(String textToAdd) throws FileNotFoundException {
+        try {
+            FileWriter fw = new FileWriter(filePath, true);
+            String amendedText = textToAdd.replaceAll("\\[\\s\\]", "|0|")
```
The code for 'amendedText' is repeated in both the saveToFile and saveAllFile methods. You can consider extracting the logic into a separate method to improve code reusability.

ashleyy2444 commented on src/main/java/venus/TaskList.java:
```diff
@@ -0,0 +1,155 @@
+        case ALL:
+            Ui.printList(this);
+            break;
+        case MARK:
```
To improve readability, you can consider separating the logic for handling different types of tasks into different methods, making each method responsible for a specific task.

ashleyy2444 commented on src/main/java/venus/Storage.java:
```diff
@@ -0,0 +1,136 @@
+import java.util.Scanner;
+
+public class Storage {
+    private String filePath;
```
Instead of working with file paths as strings, you can consider using Path for better abstraction and manipulation.
```suggestion
    Path filePath = Paths.get("data", "venus.txt");
```

ashleyy2444 reviewed (commented):
I enjoyed reading your code! Overall, your code was easy to understand. Even though some parts could have been made more clear, such as splitting longer methods into shorter ones, I would say that your code is quite easy to read. There was very good use of OOP in the code. Also, other than one of the Java coding standards being violated, you followed the them for the rest of your code! Good job! LGTM.

peterXGD committed:
Update README.md

MayDalag committed:
Check style commit for duke

MayDalag committed:
Merge branch 'branch-A-CheckStyle'

* branch-A-CheckStyle:
  Check style commit for duke

peterXGD committed:
Update README.md

MayDalag committed:
Level 10 complete

This is done to fufil requirement for level-10 for ip.

MayDalag committed:
Change code to meet the nus-se-bot feedback

Some methods do not meet the nus-se-bot requirements.

The problems pointed out are changed
to meet the code quality standard required by school.

Methods with problems are changed to have a better code quality.

Let's
* complete A-assertion branch
* complete A-CodeQuality branch

MayDalag committed:
Add assertion to DialogBox and MainWindow class

We need assertion to check if resources are loaded correctly in our code.

Assertion present in code allows us to know if our code results in any
error prematurely and with a convenient message to point to where
the error is.

Let's add assert to the two new FXML bridging classes we add in
Level 10 to check for important assertions in our code!

Other sources of exception are handled by earlier code.

To write better comments, visit oracle to find out more!

peterXGD committed:
Merge pull request #2 from peterXGD/branch-A-Assertions

Add assertion to DialogBox and MainWindow class

MayDalag committed:
Add assertion to DialogBox and MainWindow class

We need assertion to check if resources are loaded correctly in our code.

Assertion present in code allows us to know if our code results in any
error prematurely and with a convenient message to point to where
the error is.

Let's add assert to the two new FXML bridging classes we add in
Level 10 to check for important assertions in our code!

Other sources of exception are handled by earlier code.

To write better comments, visit oracle to find out more!

MayDalag committed:
Fix code quality issues

There are some code quality issues spotted by bot.

Making the code better allows us to understand the
code better and make it easier to change or
handle bugs.

Let's handle the issues spotted by bot and also
issues like magic numbers.

MayDalag committed:
Merge branch 'branch-A-CodeQuality'

* branch-A-CodeQuality:
  Fix code quality issues

MayDalag committed:
BCD Extension - C-DetectDuplicates

As required, a C-DetectDuplicate extension is is added.

MayDalag committed:
Update Ui.png for docs

MayDalag committed:
Add user guide for this duke project

This is done to fufil grading requirement.

MayDalag committed:
Latest Release v0.2 source code.

The final v0.2 release Jar is uploaded with last minute changes that differ from the previous releases.

MayDalag committed:
Update file to make it work correctly on Windows

Some commands need to be parsed differently to work on both mac and windows.

Parse.java and storage file are modified to load and save list correctly.

peterXGD committed:
Update README.md

MayDalag committed:
Change readme to show table correctly

MayDalag committed:
Merge branch 'master' of https://github.com/peterXGD/ip

* 'master' of https://github.com/peterXGD/ip:
  Update README.md
